### ISSUE TITLE
HPA does not scale down when it fail to fetch external data

### ISSUE THREAD

**Issue by mengschin** [kubernetes/kubernetes#109214]
**What happened?**
I have a HPA which use datadog metric as metric input, and the metric is capturing the request count for an endpoint. The HPA scale out the pod(s) as expected when there is request count receive. However, I notice the scale down doesn't happen when there is no metric value received for the HPA.

Sample HPA and Warning raised:
Behavior: The current target average value is stay at 16834m when there is no http request receive on the datadog's endpoint. 

```
Name:                                                                                               sample-autoscaler
Namespace:                                                                                          core
Labels:                                                                                             <none>
Annotations:                                                                                        <none>
CreationTimestamp:                                                                                  Tue, 15 Mar 2022 11:54:35 +0800
Reference:                                                                                          Deployment/sample-autoscaler-deployment
Metrics:                                                                                            ( current / target )
  "Sample.test.get.count" (target average value):  				                       16834m / 15
Min replicas:                                                                                       1
Max replicas:                                                                                       15
Deployment pods:                                                                                    15 current / 15 desired
Conditions:
  
Events:
  Type     Reason                       Age                    From                       Message
  ----     ------                       ----                   ----                       -------
 Warning  FailedComputeMetricsReplicas  108m (x12 over 111m)  horizontal-pod-autoscaler  invalid metrics (1 invalid out of 1), first error is: failed to get Sample.test.get.count external metric: unable to get external metric core/Sample.test.get.count/&LabelSelector{},MatchExpressions:[]LabelSelectorRequirement{},}: unable to fetch metrics from external metrics API: the server is currently unable to handle the request (get Sample.test.get.count.external.metrics.k8s.io)
 
 Warning  FailedGetExternalMetric       92s (x441 over 111m)  horizontal-pod-autoscaler  unable to get external metric core/Sample.test.get.count/&LabelSelector{}: unable to fetch metrics from external metrics API: the server is currently unable to handle the request (get Sample.test.get.count.external.metrics.k8s.io)

```

Question: 
It seems to me the HPA will only scale down when there is a constant metric value fetched from datadog.
Does HPA has an option to scale down the replica to Min replicas when it fail to fetch external data (**_FailedComputeMetricsReplicas_** or **_FailedGetExternalMetric_**)?

**Comment by pacoxu** [comment]
similar to #79365 and #106247
/cc @mwielgus

**Comment by mengschin** [comment]
@pacoxu 
Is this similar to the issues you tagged?

**Comment by pacoxu** [comment]
> Is this similar to the issues you tagged?

After re-read the description, it seems no similar issues. 

> Does HPA has an option to scale down the replica to Min replicas when it fail to fetch external data (**_FailedComputeMetricsReplicas_** or **_FailedGetExternalMetric_**)?

I think the current behavior is expected.
There is no such option as I know. There are no further attribute in https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#metrictarget-v2beta2-autoscaling.

**Comment by mengschin** [comment]
Thanks @pacoxu 

Just to confirm, is that the actual behavior? That, HPA wont scale down when there is no metric value received?

**Comment by jesseduffield** [comment]
I'm having this same problem, also using datadog for capturing request latency. My pods scale up as expected but when no more requests come in, they pods fail to scale down. I'd like to be able to configure an option to have my pods scale down when there's been no more metrics after a period of time.

**Comment by jesseduffield** [comment]
To flesh out my above comment there's a few possible solutions to this that come to mind:
1) (outside k8s) somehow have the external metric report a fallback value if none is present, which would lead to scaling down to one pod
2) in k8s specify a fallback value when metric is missing, on per metric basis. Might need to define a time range for when to stop using last value
3) in k8s define a fallback metric to use when parent metric is missing
4) use a separate metric e.g. cpu alongside the request latency metric, but specify that if one metric is missing we're allowed to scale down based on the other metric alone.

**Comment by JRemitz** [comment]
I happened upon this issue for similar reasons. We're using hpa `v2` in Kubernetes `1.23`.  We have two autoscaling metrics.  One is external to datadog, the other is usage for cpu.  When the external metric was unavailable, the status of the HPA was:

```
the HPA was unable to compute the replica count: unable to get external
      metric _____ unable
      to fetch metrics from external metrics API: Internal error occurred: DatadogMetric
      is invalid, err: Invalid metric (from backend), query: ____
```

The HPA status showed the TARGETS as: `<unknown>/500m (avg), 89%/80%` - the first being the external metric, the second is the CPU metric. The CPU however, was not pegged at 89%.  Seemingly this was frozen.  I removed the external metric from the criteria and the CPU dropped to `0%/80%` and the HPA scaled down.

Is this expected behavior? I can't figure out why this was stuck.
